update proxy documentation (#357)
CartoDB/carto-selfhosted-helm: customizations/README.md

### Configure external proxy

> CARTO self-hosted provides support for operating behind an HTTP proxy. The proxy acts as a gateway, enabling CARTO self-hosted components to establish connections with essential external services like Google APIs, Mapbox, and others.

Reviewer: ```suggestion
CARTO self-hosted provides support for operating behind an **HTTP** proxy. The proxy acts as a HTTP gateway, enabling CARTO self-hosted components to establish connections with essential external services like Google APIs, Mapbox, and others.
```

> CARTO self-hosted provides support for operating behind an HTTP proxy. The proxy acts as a gateway, enabling CARTO self-hosted components to establish connections with essential external services like Google APIs, Mapbox, and others.

Reviewer: ```suggestion

> :warning: CARTO Self-hosted is not installing any proxy component, it just connects to an existing proxy software provided by the client.

CARTO self-hosted provides support for operating behind an HTTP proxy. The proxy acts as a gateway, enabling CARTO self-hosted components to establish connections with essential external services like Google APIs, Mapbox, and others.
```

> > :warning: Please note that at the moment, only HTTP proxy is supported.

Reviewer: ```suggestion
> :warning: Please note that at the moment, **only HTTP protocol** proxy configuration is supported
```

You can use this [customizations](../customizations/proxy/) file as an example.

> The externalProxy.excludedDomains property allows you to specify a comprehensive list of domains that should not be proxied. This is useful in scenarios where you want to exclude certain services, such as internal Redis or Postgresql, from being routed through the proxy.

Reviewer: ```suggestion
The ``externalProxy.excludedDomains`` property allows you to specify a comprehensive list of domains that should not be proxied. This is useful in scenarios where you want to exclude certain services, such as internal Redis or Postgresql, from being routed through the proxy.
```

> The externalProxy.excludedDomains property allows you to specify a comprehensive list of domains that should not be proxied. This is useful in scenarios where you want to exclude certain services, such as internal Redis or Postgresql, from being routed through the proxy.

Reviewer: ```suggestion
The externalProxy.excludedDomains property allows you to specify a comprehensive list of domains that should not be proxied. 
```

I've removed the redis / postgresql example because they use protocols that are not supported by the proxy AFAIK (port 5432 and port 6379).

> > :warning: Please be aware that your proxy user/password credentials will be exposed in clear text within the environment variables of the CARTO components.

Reviewer: ```suggestion
> :warning: If the proxy requires user / password authentication, please be aware that both user/password credentials will be exposed in clear text within the environment variables of the CARTO components.
```

#### Supported datawarehouses

Note that while certain data warehouses can be configured to work with the proxy, there are others that will inherently bypass it. Therefore, if you have a restrictive network policy in place, you will need to explicitly allow this egress non-proxied traffic.

 | Datawarehouse | Proxy support |
 | ------------- | ------------- |
 | BigQuery      | Yes           |
 | Snowflake     | Yes           |
 | Databricks    | No            |
 | Postgres      | No            |
 | Redshift      | No            |

 > :warning: There's no need to include the non supported datawarehouses in the `externalProxy.excludedDomains` list. CARTO self-hosted components will automatically attempt a direct connection to those datawarehouses.

#### Enhaced control over non-proxied egress traffic

When no network policy is enforced, all outgoing traffic that does not pass through a proxy will be permitted.

> In restrictive environments, it is important to maintain strict control over connections made by CARTO self-hosted components. To achieve this, you should configure your proxy to allow only approved external services (whitelisting), while blocking any other outgoing traffic that does not go through the proxy. A comprehensive list of domains that must be whitelisted by the proxy for proper functioning of CARTO self-hosted can be found [here](../customizations/proxy/config/whitelisted_domains).

Reviewer: ```suggestion
In restrictive environments, it is important to maintain strict control over connections made by CARTO self-hosted components. To achieve this, you should configure your proxy to allow only approved external services (whitelisting), while blocking any other outgoing traffic that does not go through the proxy. A comprehensive list of domains that must be whitelisted by the proxy for the proper functioning of CARTO self-hosted can be found [here](../customizations/proxy/config/whitelisted_domains).
```
